include/libdnf/base/transaction.hpp

```diff
@@ -86,14 +87,14 @@ class Transaction {
     /// @param callbacks Callbacks to be called during rpm transaction.
     /// @param description Description of the transaction (the console command for CLI,
     //                     verbose description for API usage)
-    /// @param user_id UID of the user that started the transaction.
     /// @param comment Any comment to store in the history database along with the transaction.
+    /// @param user_id UID of the user that started the transaction.
     /// @return An enum describing the result of running the transaction.
     TransactionRunResult run(
         std::unique_ptr<libdnf::rpm::TransactionCallbacks> && callbacks,
         const std::string & description,
-        const std::optional<uint32_t> user_id,
-        const std::optional<std::string> comment);
+        const std::string & comment = {},
+        const uint32_t user_id = libdnf::INVALID_UID);
```

Comment:
Using `libdnf::INVALID_UID` is somehow misleading. Unset vs. unable to detect - shall the code make a difference?

Reply:
Another way is to add a setter for `user_id` in the transaction class, but it is used only in this method. I don't like defining another constant for unused UID as I am thinking about it. What about creating an overload - one with user_id, one without it and internally it could call the same private method with `std::optional` parameter?

Reply:
Overloaded method sounds like a good option here.